CMakeLists.txt

```diff
@@ -38,7 +38,9 @@ endif()
 # use SSE for floating-point operations to avoid issues with improper fp-rounding and loss of precision
 # when moving fp-data to incompatible or less-precise registers/storage locations
 # see https://gcc.gnu.org/wiki/FloatingPointMath and https://gcc.gnu.org/wiki/x87note
+IF(NOT (CMAKE_SYSTEM_PROCESSOR MATCHES "^arm" OR  CMAKE_SYSTEM_PROCESSOR MATCHES "^aarch64") )
 add_compile_options(-mfpmath=sse -msse2)
+ENDIF(NOT (CMAKE_SYSTEM_PROCESSOR MATCHES "^arm" OR  CMAKE_SYSTEM_PROCESSOR MATCHES "^aarch64") )
```

Comment:
Please use lowercase if\endif and indent the contained block, to be consistent with the style of the rest of the build system.

Reply:
@JosiahWI Yeah using lowercase to match coding styling would be a trivial thing to do.
However, @sfan5  has suggested an alternative which doesn't require writing a script (and uses lower-case to match style).

The question would be which is the correct CMakeLists.txt file to put it.
```
/CMakeLists.txt
/source/Irrlicht/CMakeLists.txt
```
-----

Diff from @sfan5
See https://github.com/minetest/minetest/pull/11853#issuecomment-991896648

```diff
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 4803b475b..c3fec8822 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -759,6 +759,15 @@ else()
                endif()
        endif()
 
+       # Enable SSE for floating point math on 32-bit x86 by default
+       # reasoning see issue #11810 and https://gcc.gnu.org/wiki/FloatingPointMath
+       if(CMAKE_SIZEOF_VOID_P EQUAL 4)
+               check_c_source_compiles("#ifndef __i686__\n#error\n#endif\nint main(){}" IS_I686)
+               if(IS_I686)
+                       set(OTHER_FLAGS "${OTHER_FLAGS} -mfpmath=sse -msse")
+               endif()
+       endif()
+
        if(MINGW)
                set(OTHER_FLAGS "${OTHER_FLAGS} -mthreads -fexceptions")
                set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DWIN32_LEAN_AND_MEAN")
```

Reply:
If we put this in /CMakeLists.txt it will be clear that it applies to every target after it is defined, so I think that is a good place to put it. If it were put in /source/Irrlicht/CMakeLists.txt for example, I believe it may or may not affect my unit test work at all depending on which subdirectory gets added first.

Reply:
How should the flags be set?
In that diff the following is used:
```
set(OTHER_FLAGS "${OTHER_FLAGS} -mfpmath=sse -msse")
```

I guess it would be good to use this?
```
add_compile_flags(-mfpmath=sse -msse)
```
Or is there a specfic convention for setting compiler flags in irrlicht (like with minetest)

Reply:
https://github.com/minetest/irrlicht/blob/master/source/Irrlicht/CMakeLists.txt#L20

Reply:
So that should be in source/Irrlicht/CMakeLists.txt

I added it just after the if block at line 20:
```
if(CMAKE_CXX_COMPILER_ID MATCHES "^(GNU|Clang|AppleClang)$")
	set(CMAKE_CXX_FLAGS_RELEASE "-O3")
	set(CMAKE_CXX_FLAGS_DEBUG "-g")

	add_compile_options(-Wall -pipe -fno-exceptions -fno-rtti)
elseif(MSVC)
	string(APPEND CMAKE_CXX_STANDARD_LIBRARIES " msvcrt.lib") # ???? fuck off

	add_compile_options(/GR- /Zl)
endif()

# Enable SSE for floating point math on 32-bit x86 by default
# reasoning see minetest issue #11810 and https://gcc.gnu.org/wiki/FloatingPointMath
if(CMAKE_SIZEOF_VOID_P EQUAL 4)
	check_c_source_compiles("#ifndef __i686__\n#error\n#endif\nint main(){}" IS_I686)
	if(IS_I686)
		message(STATUS "Detected Intel x86: using SSE instead of x87 FPU")
		add_compile_flags(-mfpmath=sse -msse)
	endif()
endif()
```

On Debian Buster the version of cmake is 3.13 (but 3.18 is available from backports).

It complained that **check_c_source_compiles** was an unknown command.
Is there something that needs to be set in order for cmake to use **check_c_source_compiles**

Reply:
You may need to include the CheckCSourceCompiles module first.

https://cmake.org/cmake/help/v3.5/module/CheckCSourceCompiles.html

Reply:
I neeed to add this as well:
```
enable_language(C)
include(CheckCSourceCompiles)
```

But now it complains that **add_compile_options** is an unknown command.
does that need an include or other command to enable?

Reply:
I needed to add the following:

```
enable_languag(C)
include(CheckCSourceCompiles)
```

Also add_compile_flags should have been add_compile_options